### PR TITLE
Implement setFlagForAnnotations

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**

### CODE REVIEW

File: ios/RNTPTDocumentView.m [hunk 1]
```diff
@@ -876,6 +876,92 @@ - (void)saveDocumentWithCompletionHandler:(void (^)(NSString * _Nullable filePat
...
+                } else if ([flag isEqualToString:@"print"]) {
+                    annotFlag = e_ptprint;
+                } else if ([flag isEqualToString:@"read_only"]) {
+                    annotFlag = e_ptread_only;
+                } else if ([flag isEqualToString:@"toggle_no_view"]) {
+                    annotFlag = e_pttoggle_no_view;
+                }
+                
+                if (annotFlag != -1) {
+                    [annot SetFlag:annotFlag value:flagValue];
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:946]: This line appears to be an issue, where `flagValue` is an NSNumber object, but the method is expecting a BOOL. If you change it to `[flagValue boolValue]`, it should work.

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]
Missing enum in Config.js, see `FieldFlags` for example

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -1689,6 +1691,70 @@ public boolean handleBackButton() {
...
+                String annotId = annotFlagData.getString(KEY_annotId);
+                int pageNum = annotFlagData.getInt(KEY_annotPage);
+                String flag = annotFlagData.getString(KEY_annotFlag);
+                boolean flagValue = annotFlagData.getBoolean(KEY_annotFlagValue);
+
+                Annot annot = ViewerUtils.getAnnotById(pdfViewCtrl, annotId, pageNum);
+                int flagVal = -1;
+                if (annot != null && annot.isValid() && flag != null) {
+                    switch (flag) {
+                        case "hidden":
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:1716]: constant out in static final variable

File: ios/RNTPTDocumentView.m [hunk 3]
```diff
@@ -876,6 +876,92 @@ - (void)saveDocumentWithCompletionHandler:(void (^)(NSString * _Nullable filePat
...
+                }
+
+                int annotFlag = -1;
+                if ([flag isEqualToString:@"hidden"]) {
+                    annotFlag = e_pthidden;
+                } else if ([flag isEqualToString:@"invisible"]) {
+                    annotFlag = e_ptinvisible;
+                } else if ([flag isEqualToString:@"locked"]) {
+                    annotFlag = e_ptlocked;
+                } else if ([flag isEqualToString:@"locked_contents"]) {
```
Comment (sgong-pdftron) [inline ios/RNTPTDocumentView.m:929]: update the strings to match config strings

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -751,6 +757,52 @@ this._viewer.handleBackButton().then((handled) => {
...
 
+##### selectAnnotation
+To select the specified annotation in the current document.
+
+Parameters:
+
+Name | Type | Description
+--- | --- | ---
+id | string | the id of the target annotation
+pageNumber | int | the page number where the targe annotation is located
```
Comment (sgong-pdftron) [inline README.md:768]: mention page is 1-indexed

File: README.md [hunk 5]
```diff
@@ -534,6 +534,9 @@ Name | Type | Description
 action | string | the action that occurred (add, delete, modify)
 annotations | array | array of annotation data in the format `{id: string, pageNumber: number}`
 
+##### annotationPermissionCheckEnabled
+bool, optional, default to false If true, annotation created by user A cannot be modified by user B, else anyone can modify any annotation.
```
Comment (sgong-pdftron) [inline README.md:538]: if true should go new line, you can click on your branch on github to check how it looks

Comment (sgong-pdftron) [reply README.md:538]: change to 

If true, annotation's flags will be taken into account when it is selected, for example, a locked annotation can not be resized or moved.

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 6]
```diff
@@ -1693,6 +1706,68 @@ public boolean handleBackButton() {
...
+                        case kEY_flagPrint:
+                            flagVal = Annot.e_print;
+                            break;
+                        case KEY_flagReadOnly:
+                            flagVal = Annot.e_read_only;
+                            break;
+                        case KEY_flagToggleNoView:
+                            flagVal = Annot.e_toggle_no_view;
+                    }
+                    annot.setFlag(flagVal, flagValue);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:1761]: call pdfviewctrl.update(annot, page) to update in the viewer

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (sgong-pdftron) [reply README.md:546]: yes fix all, if true should always be on new line

**Review by sgong-pdftron — APPROVED** [review]

**Comment by sgong-pdftron** [comment]
Android OK

File: ios/RNTPTDocumentView.m [hunk 7]
```diff
@@ -876,6 +876,92 @@ - (void)saveDocumentWithCompletionHandler:(void (^)(NSString * _Nullable filePat
...
+{
+    if (annotationFlagList.count == 0) {
+        return;
+    }
+    
+    PTPDFViewCtrl *pdfViewCtrl = self.pdfViewCtrl;
+    BOOL shouldUnlock = NO;
+    @try {
+        [pdfViewCtrl DocLock:YES];
+        shouldUnlock = YES;
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:891]: This way of locking is correct, but it's the "old way" of doing it, and we're not writing new code like this anymore. We have a couple of convenience methods that make this easier to write here (one that acquires a read lock and one that acquires a write lock): https://www.pdftron.com/api/ios/Classes/PTPDFViewCtrl.html#/Locking

You can see an example of its usage in the `deleteAnnotations:` method in this file. Could you please change any new code you've written to use the block-based locking? Thank you!

**Review by JamesTron — CHANGES_REQUESTED** [review]

**Review by JamesTron — APPROVED** [review]
Looks good 👍

**Pull request merged**